### DISCUSSION TITLE
Trex package manager + Nest.land module registry for Deno = ❤

### DISCUSSION THREAD

**Discussion by buttercubz** [nestdotland/nest.land#219] [Show and tell]
If it's coming from the js node environment, chances are it's related to npm, now npm is a private company with a centralized package log, and if you have time you know of issues that have occurred like [left-pad](https://blog.npmjs.org/post/141577284765/kik-left-pad-and-npm) or [eventStream.](https://blog.npmjs.org/post/180565383195/details-about-the-event-stream-incident)

The `nest.land` team is creating a package record for the deno environment, which avoids creating problems like the ones mentioned above. `nest.land` is powered by blockchain technology, which provides an immutable registry, it has a system to report malicious packages, all a safe and reliable environment.

this is where `Trex` comes in, `Trex` is a package manager for deno based on the import map feature of owns deno, the purpose of Trex is to make dependency management easier, Trex takes the way that npm works (only the commands) to be more friendly and easy to use.

#### install Trex

from `nest.land`

```sh
$  deno install -A --unstable -n trex https://x.nest.land/Trex@1.1.0/cli.ts
```

or from `deno.land/x/trex`

```sh
$  deno install -A --unstable -n trex https://deno.land/x/trex/cli.ts
```

#### how does it work?

you write the following in the terminal.

```sh
$ trex install --map oak 
```
When using the `--map` flag, trex will try to install the oak package, `--map` is to install packages hosted in `deno.land/std` and `deno.land/x`

> **note:** you can install a specific version using @ example `$ trex i --map oak@v6.0.0`

to install packages hosted on `nest.land` just use the `--nest` flag.

```sh
$ trex install --nest MandarineTS@1.2.1
```
just type in the package name and specify the version using `@`

if the package you need is only hosted in a repository, you can use the `--pkg` flag to install it.

```sh
$ trex install --pkg oakserver/oak/mod.ts oak
```

It works in the following way.

```sh
$ trex install --pkg [user]/[repo or repo@tag]/[path/to/file] [packageName]
```

To see the complete usage guide visit the official repository [here](https://github.com/crewdevio/Trex)

**Comment by t8** [comment]
I think that what you guys have built is a really interesting concept on top of how Deno imports code. Introducing a clean way to vendor and manage modules is something that I think a lot more people will appreciate in the coming time.

In addition, I really respect that you all have rolled out support for multiple registries instead of mandating one in particular. 
This adds to the decentralized ideals of Deno, and we're honored to be supported!
